lib/shopify-cli/environment.rb

```diff
@@ -34,6 +34,10 @@ def self.partners_domain(env_variables: ENV)
       end
     end
 
+    def self.use_spin?(env_variables: ENV)
+      !env_variables[Constants::EnvironmentVariables::SPIN_WORKSPACE].nil?
```

Comment:
Including SPIN_NAMESPACE would likely make sense because yes the spin URL won't be correct if it isn't also set.

SHOPIFY_APP_CLI_SPIN_PARTNERS couldn't be used though because that will cause issues in this case because want to bypass the partners proxy and push directly to script service. We essentially have the following configurations and ENV variables:

**Using partners proxy in spin**
SPIN_NAMESPACE=danwd-king
SPIN_WORKSPACE=abcd
SHOPIFY_APP_CLI_SPIN_PARTNERS=1

**Bypassing partners proxy in spin**
SPIN_NAMESPACE=danwd-king
SPIN_WORKSPACE=abcd
BYPASS_PARTNERS_PROXY=1

**Local dev with partners**
SHOPIFY_APP_CLI_LOCAL_PARTNERS=1

**Bypassing partners proxy locally**
BYPASS_PARTNERS_PROXY=1

I think we should likely clean up some of these and maybe deprecate the local partners but that specific environment variable is called out in one of the CLI's readme files and is used for some of the identity calls and might be used outside of the scripts team.

Reply:
Ah okay gotcha I agree, that makes sense. Thanks for the clarification!

> Including SPIN_NAMESPACE would likely make sense because yes the spin URL won't be correct if it isn't also set.

Cool, if you add that then I'm good with this 👍

Reply:
Added